### ISSUE TITLE
Add support for peerDependenciesMeta

### ISSUE THREAD

**Issue by mmanciop** [projen/projen#2660]
Projen should have support for [`peerDependenciesMeta`](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#peerdependenciesmeta):

> When a user installs your package, npm will emit warnings if packages specified in peerDependencies are not already installed. The peerDependenciesMeta field serves to provide npm more information on how your peer dependencies are to be used. Specifically, it allows peer dependencies to be marked as optional.

**Comment by mrgrain** [comment]
Agreed. In the meantime people can use `project.package.addField()` or [overrides](http://projen.io/escape-hatches.html#overrides).

**Comment by mmanciop** [comment]
Or, if you have something **_compelling_** you to add the `peerDependenciesMeta` right after `peerDependencies`, you can do a horrible hack like https://github.com/lumigo-io/lumigo-cdk-constructs/blob/02e42f01221caa36e35ba28a16292c6090ee7acc/.projenrc.js#L37 :P
